dshop_react/src/pages/Admin/Category/CategoryImageUpload.jsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import { message, Upload } from "antd";
+import { imagesEndpoint } from "../../../services/imageService";
+import api from "../../../services/api";
+
+const beforeUpload = (file) => {
+  const isJpgOrPng = file.type === "image/jpeg" || file.type === "image/png";
+  if (!isJpgOrPng) {
+    message.error("You can only upload JPG/PNG file!");
+  }
+  const isLt2M = file.size / 1024 / 1024 < 2;
+  if (!isLt2M) {
+    message.error("Image must smaller than 2MB!");
+  }
+  return isJpgOrPng && isLt2M;
+};
+const CategoryImageUpload = (props) => {
+  const uploadImage = async (options) => {
```

Comment:
This logic when processing and uploading the file also needs to be used for the product. Therefore, you should write this code generically and put it in a class so that it can be used for many different components.